modules/core/src/convert.cpp

```diff
@@ -4585,7 +4585,7 @@ cvtScaleHalf_<float, short>( const float* src, size_t sstep, short* dst, size_t
                     __m128i v_dst = _mm_cvtps_ph(v_src, 0);
 
                     _mm_storel_epi64((__m128i *)(dst + x), v_dst);
-#elif defined __GNUC__ && (defined __arm__ || defined __aarch64__)
+#elif defined __GNUC__ && (defined CV_NEON || defined __aarch64__)
```

Comment:
@ilya-lavrenov Right. But there is runtime (logic) error also, it is just a test.
Could you add some links with information about compiler requirements for float16x4_t ?
```
/opt/android/android-ndk-r9/toolchains/arm-linux-androideabi-4.8/prebuilt/linux-x86_64/bin/arm-linux-androideabi-g++ -mfpu=neon -mfp16-format=ieee -mfpu=neon-fp16 -dM -E - <<< "" | grep ARM
#define __ARM_SIZEOF_WCHAR_T 32
#define __ARM_ARCH_ISA_ARM 1
#define __ARMEL__ 1
#define __ARM_ARCH_5TE__ 1
#define __ARM_FP 14
#define __ARM_NEON_FP 6
#define __ARM_SIZEOF_MINIMAL_ENUM 4
#define __ARM_PCS 1
#define __ARM_FP16_FORMAT_IEEE 1
#define __ARM_FEATURE_QBIT 1
#define __ARM_FEATURE_CLZ 1
#define __ARM_ARCH_ISA_THUMB 1
#define __ARM_ARCH 5
#define __ARM_EABI__ 1
#define __ARM_FEATURE_DSP 1
```
Error log is [here](http://pullrequest.opencv.org/buildbot/builders/precommit_pack_android/builds/107/steps/build%20sdk/logs/stdio)